varats-core/varats/report/hot_functions_report.py

```diff
@@ -0,0 +1,98 @@
+import typing as tp
+from dataclasses import dataclass
+from pathlib import Path
+
+from pandas import read_csv
+
+from varats.experiment.workload_util import WorkloadSpecificReportAggregate
+from varats.report.report import BaseReport, ReportAggregate
+
+
+@dataclass
+class XRayFunctionWrapper:
+    name: str
+    count: int
+    sum_time: float
+
+
+class HotFunctionReport(BaseReport, shorthand="HFR", file_type=".csv"):
+
+    MAX_TRACK_FUNCTIONS = 50
+
+    def __init__(self, path: Path) -> None:
+        super().__init__(path)
+        self.__function_data = read_csv(path)
+
+    def top_n_functions(self, limit=10) -> tp.List[XRayFunctionWrapper]:
+        self.__function_data.sort_values(
+            by='sum', ascending=False, inplace=True
+        )
+        return [
+            XRayFunctionWrapper(
+                name=row["function"], count=row['count'], sum_time=row["sum"]
+            ) for _, row in self.__function_data.head(limit).iterrows()
+        ]
+
+    def hot_functions(self, threshold=2) -> tp.List[XRayFunctionWrapper]:
+        """
+        Args:
+            threshold: min percentage a function needs as self time to count as hot
+        """
+        if threshold < 0 or threshold > 100:
+            raise ValueError(
+                "Threshold value needs to be in the range [0,...,100] "
+                f"but was {threshold}"
+            )
+
+        self.__function_data.sort_values(
+            by='sum', ascending=False, inplace=True
+        )
+        # The total time tracked only includes time spend in the top n
+        # (MAX_TRACK_FUNCTIONS) functions
+        total_time_tracked = self.__function_data["sum"].sum()
```

Comment:
This currently calculates the sum of all measured functions' execution times including their children and, thus, gives a number much larger than the total execution time.
To get the correct total time, you should use `max()` instead.

```suggestion
        total_time_tracked = self.__function_data["sum"].max()
```